docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md

```diff
@@ -1 +1,55 @@
+
+When processing time is around 500 FLOPS (floating point operations) or more, then the two-thread rule does not apply to **ConcurrentQueue&lt;T&gt;**, which then has very good scalability. **Queue&lt;T&gt;** does not scale well in this scenario.
+
+In mixed producer-consumer scenarios, when the processing time is very small, a **Queue&lt;T&gt;** that has an external lock scales better than **ConcurrentQueue&lt;T&gt;** does. However, when processing time is around 500 FLOPS or more, then the **ConcurrentQueue&lt;T&gt;** scales better.
+
+## ConcurrentStack vs. Stack
+
+In pure producer-consumer scenarios, when processing time is very small, then [System.Collections.Concurrent.ConcurrentStack&lt;T&gt;](http://dotnet.github.io/api/System.Collections.Concurrent.ConcurrentStack%601.html) and [System.Collections.Generic.Stack&lt;T&gt;](http://dotnet.github.io/api/System.Collections.Generic.Stack%601.html) that has an external lock will probably perform about the same with one dedicated pushing thread and one dedicated popping thread. However, as the number of threads increases, both types slow down because of increased contention, and **Stack&lt;T&gt;** might perform better than **ConcurrentStack&lt;T&gt;**. When processing time is around 500 FLOPS or more, then both types scale at about the same rate. 
+
+In mixed producer-consumer scenarios, **ConcurrentStack&lt;T&gt;** is faster for both small and large workloads.
+
+The use of the **PushRange** and **TryPopRange** may greatly speed up access times.
+
+## ConcurrentDictionary vs. Dictionary
+
+In general, use a [System.Collections.Concurrent.ConcurrentDictionary&lt;TKey, TValue&gt;](http://dotnet.github.io/api/System.Collections.Concurrent.ConcurrentDictionary%602.html) in any scenario where you are adding and updating keys or values concurrently from multiple threads. In scenarios that involve frequent updates and relatively few reads, the **ConcurrentDictionary&lt;TKey, TValue&gt;** generally offers modest benefits. In scenarios that involve many reads and many updates, the **ConcurrentDictionary&lt;TKey, TValue&gt;** generally is significantly faster on computers that have any number of cores.
+
+In scenarios that involve frequent updates, you can increase the degree of concurrency in the **ConcurrentDictionary&lt;TKey, TValue&gt;** and then measure to see whether performance increases on computers that have more cores. If you change the concurrency level, avoid global operations as much as possible.
+
+If you are only reading key or values, the [Dictionary&lt;TKey, TValue&gt;](http://dotnet.github.io/api/System.Collections.Generic.Dictionary%602.html) is faster because no synchronization is required if the dictionary is not being modified by any threads.
```

Comment:
I like the idea of using the fully qualified name on first mention, as it reinforces memory of what namespace it's in, especially in the case where we have several similarly named namespaces. I've also been following the convention of linking on first mention, then bolding further mentions.

We should discuss at our next meeting and establish style guidelines for consistency.